imagery-cesium/plugins/draw-annotations.visualizer.ts

```diff
@@ -0,0 +1,223 @@
+					showMeasures: feature.properties.showMeasures || false,
+					showArea: feature.properties.showArea || false,
+					label: feature.properties.label || {
+						text: "",
+						geometry: null,
+					},
+					icon: feature.properties.icon || "",
+					undeletable: feature.properties.undeletable || false,
+					labelSize: feature.properties.labelSize || 28,
+					labelTranslateOn: feature.properties.labelTranslateOn || false,
+				};
+			}
+		);
+	}
+
+	startDrawing(mode: AnnotationMode): Observable<FeatureCollection<GeometryObject>> {
+		this.drawingMode = mode;
+		this.mapEventsSubscription.unsubscribe();
+		this.mapEventsSubscription = new Subscription();
+
+		if (this.drawingMode === AnnotationMode.Point) {
+			return this.leftClickEvent$.pipe(take(1), map((screenPixels: IPixelPosition) => {
+				const earthPosition = this.cesiumMap.getEarthPositionFromScreenPixels(screenPixels.position);
+				return this.generateGeometry(this.drawingMode, [earthPosition]);
+			}));
+		}
+
+		this.mapEventsSubscription.add(
+			this.leftClickEvent$.subscribe((screenPixels: IPixelPosition) => {
+				const earthPosition = this.cesiumMap.getEarthPositionFromScreenPixels(screenPixels.position);
+				if (defined(earthPosition)) {
+					if (this.activeShapePoints.length === 0) {
+						// floating point is the point that sticks to the cursor when drawing!
+						// on every mouse move I update it's value to the last mouse position
+						// here I create it for the first time
+						this.floatingPoint = this.addAnnotation(AnnotationType.Point, earthPosition);
+
+						this.activeShapePoints.push(earthPosition);
+						const dynamicPositions = new CallbackProperty(() => {
+							if (this.drawingMode === AnnotationMode.Polygon) {
+								return new PolygonHierarchy(this.activeShapePoints);
+							}
+							return this.activeShapePoints;
+						}, false);
+						this.activeShape = this.addAnnotation(this.drawingMode as unknown as AnnotationType, dynamicPositions);
+					}
+					this.activeShapePoints.push(earthPosition);
+				}
+			})
+		);
+		this.mapEventsSubscription.add(
+			this.mouseMoveEvent$.subscribe((event) => {
+				this.updateLastActivePoint(event.endPosition);
+			})
+		);
+
+		return this.leftDoubleClickEvent$.pipe(map(() => this.onDrawEnd()));
+	}
+
+	private updateLastActivePoint(pixelPoint: Cartesian2) {
+		if (defined(this.floatingPoint)) {
+			const newPosition = this.cesiumMap.getEarthPositionFromScreenPixels(pixelPoint);
+			if (defined(newPosition)) {
+				(this.floatingPoint.position as any).setValue(newPosition);
+				this.activeShapePoints.pop();
+				this.activeShapePoints.push(newPosition);
+			}
+		}
+	}
+
+	private addAnnotation(type: AnnotationType, positionData: Property | Cartesian3): Entity {
+		let shape: Entity;
+		switch (type) {
+			case AnnotationType.LineString: {
+				shape = this.viewer.entities.add({
+					polyline: {
+						positions: positionData as Property,
+						clampToGround: true,
+						width: 3,
```

Comment:
by annotation style

Reply:
I will do it after this pr as separate task